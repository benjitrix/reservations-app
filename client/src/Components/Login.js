import React, {useState, useContext} from 'react';
import AuthService from '../Services/AuthServices'
import { AuthContext } from '../Context/AuthContext'
import Message from './Message'

// user login component
const Login = (props) => {
  const [user, setUser] = useState({username: "", password: "", role: ""});
  const [message, setMessage] = useState(null);
  const authContext = useContext(AuthContext);

  // capture username and password from input boxes, add to array
  const onChange = (e) => {
    setUser({...user, [e.target.name] : e.target.value})
  }

  // Login for user registered user using username and password
  const onSubmit = (e) => {
    e.preventDefault();
    AuthService.login(user).then(data => {
      console.log(data)
      const { isAuthenticated, user } = data;
      if (isAuthenticated) {
        authContext.setUser(user);
        authContext.setIsAuthenticated(isAuthenticated);
        if (user.role === "user") {
          props.history.push('/reservations');
        } else {
          props.history.push("/admin");
        }
      } else {
        setMessage(message);
      }
    }) 
  }

  return (
    <div>
      <form onSubmit={onSubmit}>
      <h3>Please sign in</h3>
      <label htmlFor="username" className="sr-only">Username: </label>
      <input  type="text" 
              name="username" 
              onChange={onChange} 
              className="form-control" 
              placeholder="Enter username" />
      <input  type="password" 
              name="password" 
              onChange={onChange} 
              className="form-control" 
              placeholder="Enter password" />

      <button type="submit" className="btn btn-lg btn-primary btn-block">Log in</button>
      </form>
      {message ? <Message message={message} /> : null}
    </div>
  )
}

export default Login
