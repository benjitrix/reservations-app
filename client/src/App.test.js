import React from 'react';
import Login from './Components/Login'
import renderer from 'react-test-renderer';

// snapshot test of Login component
test('renders correctly', () => {
const tree = renderer
.create(<Login />)
.toJSON();
expect(tree).toMatchSnapshot();
});
