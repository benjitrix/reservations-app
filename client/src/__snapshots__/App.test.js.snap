// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <h3>
      Please sign in
    </h3>
    <label
      className="sr-only"
      htmlFor="username"
    >
      Username: 
    </label>
    <input
      className="form-control"
      name="username"
      onChange={[Function]}
      placeholder="Enter username"
      type="text"
    />
    <input
      className="form-control"
      name="password"
      onChange={[Function]}
      placeholder="Enter password"
      type="password"
    />
    <button
      className="btn btn-lg btn-primary btn-block"
      type="submit"
    >
      Log in
    </button>
  </form>
</div>
`;
